### PR TITLE
DAT-18147: Fossa report generation for Datical repositories

### PR DESCRIPTION
**Pull request by sayaliM0412**
feat: `.github/workflows/fossa.yml`: reusable workflow for Fossa report generation for Datical repositories

**NOTE**: Merge the PR after the report generation has been verified by the Enterprise team in jira ticket : https://datical.atlassian.net/browse/DAT-18147

### CODE REVIEW

File: .github/workflows/fossa.yml [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+        with:
+          distribution: 'temurin'
+          java-version: '17'
+
+            #look for dependencies in maven
+      - name: maven-settings-xml-action for Liquibase
+        if: inputs.org == 'liquibase'
+        uses: whelk-io/maven-settings-xml-action@v22
+        with:
+          repositories: |
```
Comment (andrulban) [inline .github/workflows/fossa.yml:64]: Hi @sayaliM0412 - looks like there are liquibase-pro and liquibase (foundation) repos, which are not related to Enterprise. Is this PR needed for DAT-18147? If it is needed, then do we need liquibase-pro and liquibase (foundation) repos part?

File: .github/workflows/fossa.yml [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+      - name: Get the commit sha
+        id: get_commit_sha
+        run: |
+          commit_sha=`(git rev-parse HEAD)`
+          echo "commit_sha=${commit_sha}" >> $GITHUB_OUTPUT
+
+      - name: Get repository name
+        id: get_repo_name
+        run: echo "repo_name=$(echo ${{ github.repository }} | cut -d'/' -f2)" >> $GITHUB_OUTPUT
```
Comment (jnewton03) [inline .github/workflows/fossa.yml:117]: can we use github's built-in vars for this instead?

File: .github/workflows/fossa.yml [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+
+      # https://docs.fossa.com/docs/download-fossa-project-attribution-reports
+      -   name: Datical- Set the dependency metadata information
+          if: inputs.org == 'datical'
+          run: |
+            sleep 120
```
Comment (jnewton03) [inline .github/workflows/fossa.yml:123]: I would remove sleep and use retry logic in the curl instead

File: .github/workflows/fossa.yml [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+            curl --location 'https://app.fossa.com/api/revisions/custom%2B40163%2Fgithub.com%2FDatical%2F${{ steps.get_repo_name.outputs.repo_name }}%24${{ steps.get_commit_sha.outputs.commit_sha }}/attribution/download?format=CSV&includeDirectDependencies=true&includeDeepDependencies=true&download=true' \
+            --header 'Authorization: Bearer ${{ secrets.FOSSA_COMBINED_REPORT }}' \
+            -o /home/runner/work/${{ steps.get_repo_name.outputs.repo_name }}/fossa_reports/${{ steps.get_repo_name.outputs.repo_name }}.csv
+            ls -l /home/runner/work/${{ steps.get_repo_name.outputs.repo_name }}/fossa_reports
+            cat /home/runner/work/${{ steps.get_repo_name.outputs.repo_name }}/fossa_reports/${{ steps.get_repo_name.outputs.repo_name }}.csv || echo "File is empty or not found"
+
+      -   name: Liquibase- Set the dependency metadata information
+          if: inputs.org == 'liquibase'
+          run: |
+            sleep 120
```
Comment (jnewton03) [inline .github/workflows/fossa.yml:134]: I would remove sleep and use retry logic in the curl instead

**Comment by sayaliM0412** [comment]
I dont think this PR is needed as we have the combine logic in datical/build-logic https://github.com/Datical/build-logic/pull/252

**Pull request merged**